groovy/postInit/electronics/ElectronicCircuits.groovy

```diff
@@ -10,38 +12,60 @@ mods.gregtech.assembler.removeByInput(30, [metaitem('wireFineCopper') * 4, metai
-        [metaitem('circuit.vacuum_tube'), metaitem('circuit_board.basic'), metaitem('circuit.vacuum_tube')],
-        [ore('cableGtSingleRedAlloy'), ore('cableGtSingleRedAlloy'), ore('cableGtSingleRedAlloy')]])
-
-crafting.replaceShaped("gregtech:electronic_circuit_mv", metaitem('circuit.good_electronic'), [
-        [metaitem('component.diode'), ore('craftingToolWireCutter'), metaitem('component.diode')],
-        [metaitem('circuit.electronic'), metaitem('circuit_board.good'), metaitem('circuit.electronic')],
-        [ore('wireGtSingleCopper'), metaitem('circuit.electronic'), ore('wireGtSingleCopper')]])
-
-crafting.addShaped('resistor_wire_anthracite', metaitem('component.resistor') * 2, [
-    [metaitem('rubber_drop'),item('minecraft:paper'),metaitem('rubber_drop')],
-    [ore('wireGtSingleCopper'),ore('dustAnthracite'), ore('wireGtSingleCopper')],
-    [null,item('minecraft:paper'),null]])
-
-crafting.addShaped('resistor_wire_fine_anthracite', metaitem('component.resistor') * 2, [
-    [metaitem('rubber_drop'),item('minecraft:paper'),metaitem('rubber_drop')],
-    [ore('wireFineCopper'),ore('dustAnthracite'), ore('wireFineCopper')],
-    [null,item('minecraft:paper'),null]])
-	
-crafting.addShaped('resistor_wire_coke', metaitem('component.resistor') * 2, [
-    [metaitem('rubber_drop'),item('minecraft:paper'),metaitem('rubber_drop')],
-    [ore('wireGtSingleCopper'),ore('dustCoke'), ore('wireGtSingleCopper')],
-    [null,item('minecraft:paper'),null]])
+mods.gregtech.assembler.recipeBuilder()
+        .fluidInputs(fluid('glue') * 100)
+        .input(new GTRecipeItemInput(carbons, 1))
+        .inputs(ore('wireFineCopper') * 4)
+        .outputs(metaitem('component.resistor') * 2)
+        .duration(160)
+        .EUt(6)
+        .buildAndRegister();
 
-crafting.addShaped('resistor_wire_fine_coke', metaitem('component.resistor') * 2, [
-    [metaitem('rubber_drop'),item('minecraft:paper'),metaitem('rubber_drop')],
-    [ore('wireFineCopper'),ore('dustCoke'), ore('wireFineCopper')],
-    [null,item('minecraft:paper'),null]])
+mods.gregtech.assembler.recipeBuilder()
+        .fluidInputs(fluid('glue') * 100)
+        .input(new GTRecipeItemInput(carbons, 1))
+        .inputs(ore('wireFineAnnealedCopper') * 4)
+        .outputs(metaitem('component.resistor') * 4)
+        .duration(160)
+        .EUt(6)
+        .buildAndRegister();        
+
+crafting.shapedBuilder()               
+        .name('resistor_wire')            
+        .output(metaitem('component.resistor') * 2)      
+        .matrix('RPR',                      
+                'WCW',                 
+                ' P ')
+        .key('R', metaitem('rubber_drop'))  
+        .key('P', item('minecraft:paper'))         
+        .key('W', ore('wireGtSingleCopper') | ore('wireFineCopper'))  
+        .key('C', ore('dustAnthracite') | ore('dustCoke') | ore('dustCarbon') | ore('dustCoal') | ore('dustCharcoal'))             
```

Comment:
Is it intentional that high purity carbon can't be used here?

Reply:
It can be used, here I just chain ore dicts and hp carbon is in dustCarbon too